workers/common/src/main/kotlin/common/OrtRunService.kt

```diff
@@ -489,4 +499,32 @@ class OrtRunService(
             ortRunRepository.update(ortRunId, issues = issues.asPresent())
         }
     }
+
+    /**
+     * For scan results regarding a [RepositoryProvenance], if this provenance has a VCS path, then filter
+     * out the [OrtScanResult] summaries that are not related to this path. These summaries typically contain
+     * findings about licenses, copyrights, code snippets and other information from the scan.
+     */
+    private fun filterScanResultSummary(
+        provenances: Collection<ProvenanceResolutionResult>,
+        scanResults: Set<ScanResult>
+    ): Set<OrtScanResult> {
+        val vcsPaths = provenances
+            ?.mapNotNull { (it.packageProvenance as? RepositoryProvenance)?.vcsInfo?.path }
+            ?.filter { it.isNotEmpty() }
+            ?.toMutableSet() ?: mutableSetOf()
+
+        // Filter out scan result summaries outside the VCS path.
+        return scanResults
+            ?.map { scanResult ->
+                scanResult.mapToOrt().let { ortScanResult ->
+                    if (scanResult.provenance is RepositoryProvenance) {
+                        ortScanResult.copy(summary = ortScanResult.summary.filterByPaths(vcsPaths))
+                    } else {
+                        ortScanResult
+                    }
+                }
+            }
+            ?.toSet().orEmpty()
+    }
```

Comment:
All in all, I propose instead:

        val vcsPaths = provenances
            .filterIsInstance<RepositoryProvenance>()
            .mapNotNull { it.vcsInfo.path.takeUnless { it.isEmpty() } }

        // Filter out scan result summaries outside the VCS path.
        return scanResults.mapTo(mutableSetOf()) { scanResult ->
            scanResult.mapToOrt().let { ortScanResult ->
                if (scanResult.provenance is RepositoryProvenance) {
                    ortScanResult.copy(summary = ortScanResult.summary.filterByPaths(vcsPaths))
                } else {
                    ortScanResult
                }
            }
        }